From 98619f1fe80d8c606ebc5b6fa50fbec70311dd0e Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Tue, 28 May 2019 16:41:37 +0800
Subject: [PATCH] init: fix avc denials for init_t

Fix:
type=AVC msg=audit(1559024138.454:31): avc:  denied  { link } for
pid=190 comm="(mkdir)" scontext=system_u:system_r:init_t:s0-s15:c0.c1023
tcontext=system_u:system_r:kernel_t:s15:c0.c1023 tclass=key permissive=1

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/init.te | 1 +
 1 file changed, 1 insertion(+)

diff --git a/policy/modules/system/init.te b/policy/modules/system/init.te
index 44e06ff..fc83705 100644
--- a/policy/modules/system/init.te
+++ b/policy/modules/system/init.te
@@ -195,6 +195,7 @@ mls_process_write_all_levels(init_t)
 mls_fd_use_all_levels(init_t)
 mls_process_set_level(init_t)
 mls_fd_share_all_levels(init_t)
+mls_key_write_all_levels(init_t)
 
 # the following one is needed for libselinux:is_selinux_enabled()
 # otherwise the call fails and sysvinit tries to load the policy
-- 
2.7.4

